scripts/Interface/ui/sans_isis/summation_settings_widget.py

```diff
@@ -0,0 +1,111 @@
+        self._setupBinningTypes()
+
+    def _setupBinningTypes(self):
+        binningTypes = [
+            'Use custom binning',
+            'Use binning from monitors',
+            'Save as event data'
+        ]
+        for binningType in binningTypes:
+            self.binningType.addItem(binningType)
+
+    def _connect_signals(self):
+        self.binningType.currentIndexChanged.connect(self._handle_binning_type_changed)
+        self.overlayEventWorkspacesCheckbox.stateChanged.connect(self._handle_overlay_ews_changed)
+        self.binningOptionsLineEdit.editingFinished.connect(self._handle_binning_options_line_edit_changed)
+
+    def _binning_type_index_to_type(self, index):
+        if index == 0:
+            return BinningType.Custom
+        elif index == 1:
+            return BinningType.FromMonitors
+        elif index == 2:
+            return BinningType.SaveAsEventData
+
+    def _handle_binning_type_changed(self, index):
+        binning_type = self._binning_type_index_to_type(index)
+        self.binningTypeChanged.emit(binning_type)
+
+    def _handle_binning_options_line_edit_changed(self):
+        # Since the text box is shared we don't
+        # know which of these was actually changed.
+        # The presenter can work it out.
+        self.binSettingsChanged.emit()
+        self.additionalTimeShiftsChanged.emit()
+
+    def _handle_overlay_ews_changed(self, state):
+        self.preserveEventsChanged.emit(state != 0)
+
+    def draw_settings(self, settings):
+        self._draw_bin_settings(settings)
+        self._draw_additional_time_shifts(settings)
+        self._draw_overlay_event_workspaces(settings)
+
+    def bin_settings(self):
+        return self.binningOptionsLineEdit.text()
+
+    def additional_time_shifts(self):
+        return self.binningOptionsLineEdit.text()
+
+    def _draw_overlay_event_workspaces(self, settings):
+        if settings.has_overlay_event_workspaces():
+            self.overlayEventWorkspacesCheckbox.setVisible(True)
+            should_be_checked = settings.is_overlay_event_workspaces_enabled()
+            set_checked_without_signal(
+                self.overlayEventWorkspacesCheckbox, should_be_checked)
+        else:
+            set_checked_without_signal(
+                self.overlayEventWorkspacesCheckbox, False)
+            self.overlayEventWorkspacesCheckbox.setVisible(False)
+
+    def _draw_bin_settings(self, settings):
+        if settings.has_bin_settings():
+            self._activate_line_edit('Custom Bin Boundaries:', settings.bin_settings)
+        elif not settings.has_additional_time_shifts():
+            self._deactivate_line_edit()
+
+    def _draw_additional_time_shifts(self, settings):
+        if settings.has_additional_time_shifts():
+            self._activate_line_edit('Additional Time Shifts:', settings.additional_time_shifts)
+        elif not settings.has_bin_settings():
+            self._deactivate_line_edit()
+
```

Comment:
the 3 above functions should have the if and else statements split into different functions. We can do the check in the presenter and therefore test it

Reply:
Moving this into the presenter requires having an if statement along these lines in the presenter (which is supposed to have a degree of independence from the view)